gnucash/gnome/gnc-plugin-budget.c

```diff
@@ -284,7 +284,7 @@ gnc_budget_gui_select_budget(QofBook *book)
     gboolean ok;
 
     dlg = GTK_DIALOG(gtk_dialog_new_with_buttons(
-                         _("Select a Budget"), NULL, GTK_DIALOG_MODAL,
+                         _("Select a Budget"), gnc_ui_get_main_window(NULL), GTK_DIALOG_MODAL,
```

Comment:
The two commands calling `gnc_budget_gui_select_budget` have access to a main_window pointer. This pointer should be passed as additional parameter to `gnc_budget_gui_select_budget`